VideoAPI/VideoApi.Domain/Room.cs

```diff
@@ -0,0 +1,65 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using VideoApi.Domain.Ddd;
+using VideoApi.Domain.Enums;
+using VideoApi.Domain.Validations;
+
+namespace VideoApi.Domain
+{
+    public class Room : Entity<long>
+    {
+        public Room(Guid conferenceId, string label, VirtualCourtRoomType type)
+        {
+            ConferenceId = conferenceId;
+            Label = label;
+            Type = type;
+            Status = RoomStatus.Created;
+            RoomParticipants = new List<RoomParticipant>();
+        }
+
+        public Guid ConferenceId { get; private set; }
+        public string Label { get; private set; }
+        public VirtualCourtRoomType Type { get; private set; }
+        public RoomStatus Status { get; private set; }
+
+        public virtual List<RoomParticipant> RoomParticipants { get; }
+
+        public void AddParticipant(RoomParticipant participant)
+        {
+            if (DoesParticipantExist(participant))
+            {
+                throw new DomainRuleException(nameof(participant), "Participant already exists in conference");
+            }
+
+            RoomParticipants.Add(participant);
+        }
+
+        public void RemoveParticipant(RoomParticipant participant)
+        {
+            if (!DoesParticipantExist(participant))
+            {
+                throw new DomainRuleException(nameof(participant), "Participant does not exist in conference");
+            }
+
+            var existingParticipant = RoomParticipants.Single(x => x.ParticipantId == participant.ParticipantId);
+
+            RoomParticipants.Remove(existingParticipant);
+        }
+
+        public void UpdateStatus(RoomStatus status)
```

Comment:
We could be clever regarding the status. When the first participant is added, the status could be updated to being 'Live' and when the last participant is removed the room can be updated to 'Closed' since that is how thee room works.

Reply:
done